packages/amplify_flutter/lib/amplify.dart

```diff
@@ -158,15 +170,20 @@ class AmplifyClass extends PlatformInterface {
           underlyingException: e.toString());
     }
     try {
+      _config = _createConfigObject(configuration);
+
       bool? res = await AmplifyClass.instance
           ._configurePlatforms(_getVersion(), configuration);
       _isConfigured = res ?? false;
+
+      configCompleter.complete(_config);
       if (!_isConfigured) {
         throw AmplifyException('Amplify failed to configure.',
             recoverySuggestion:
                 AmplifyExceptionMessages.missingRecoverySuggestion);
       }
     } on PlatformException catch (e) {
+      configCompleter.completeError(e);
```

Comment:
I believe that's also why the one test is failing in `amplify_flutter`.

Reply:
Well, in my mind this completer could be something that took the place of using the `isConfigured` getter, in which case I think it would make sense to return an exception... but perhaps for now, for our use case, it would be simpler to do as you suggest. We could be clear about it in our docs.